### PR TITLE
W-12460253: Build services only once per suite in deployment tests

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

**Comment by eze210** [comment]
--validate

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/TestServicesSetup.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import static org.apache.commons.io.FileUtils.copyDirectory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Function;
+
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TemporaryFolder;
+
+public class TestServicesSetup extends ExternalResource {
```
Comment (asanguinetti) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/TestServicesSetup.java:22]: We could add some JavaDoc to this class.

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DomainDeploymentWithELServiceTestCase.java [hunk 2]
```diff
@@ -84,15 +92,14 @@ public void disposeStaleDomains() {
     TestDomainFactory.after();
   }
 
-  @Override
-  protected File getExpressionLanguageServiceFile(File tempFolder) {
+  private static File getRealExpressionLanguageServiceFile(File tempFolder) {
     final URL weaveJarUrl = WeaveServiceProvider.class.getProtectionDomain().getCodeSource().getLocation();
-    final File defaulServiceSchedulerJarFile = getResourceFile(weaveJarUrl.getFile(), tempFolder);
+    final File defaultServiceSchedulerJarFile = getResourceFile(weaveJarUrl.getFile(), tempFolder);
```
Comment (asanguinetti) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DomainDeploymentWithELServiceTestCase.java:97]: We could fix this even further:
```suggestion
    final File defaultServiceJarFile = getResourceFile(weaveJarUrl.getFile(), tempFolder);
```

**Review by asanguinetti — CHANGES_REQUESTED** [review]

**Review by asanguinetti — APPROVED** [review]

**Comment by eze210** [comment]
--validate

**Pull request merged**